examples/lock-app/linux/src/LockAppCommandDelegate.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "LockAppCommandDelegate.h"
+#include <platform/PlatformManager.h>
+
+#include <LockManager.h>
+#include <utility>
+
+using chip::to_underlying;
+
+class LockAppCommandHandler
+{
+public:
+    static LockAppCommandHandler * FromJSON(const char * json);
+
+    static void HandleCommand(intptr_t context);
+
+private:
+    LockAppCommandHandler(std::string cmd, Json::Value params) : mCommandName(std::move(cmd)), mCommandParameters(std::move(params))
+    {}
+
+    std::string mCommandName;
+    Json::Value mCommandParameters;
+};
+
+LockAppCommandHandler * LockAppCommandHandler::FromJSON(const char * json)
+{
+    // Command format:
+    // { "Cmd": "SetDoorState", "Params": { "EndpointId": 1, "DoorState": 2} }
+    Json::Reader reader;
+    Json::Value value;
+    if (!reader.parse(json, value))
+    {
+        ChipLogError(NotSpecified, "Lock App: Error parsing JSON with error %s:", reader.getFormattedErrorMessages().c_str());
+        return nullptr;
+    }
+
+    if (value.empty() || !value.isObject())
+    {
+        ChipLogError(NotSpecified, "Lock App: Invalid JSON command received");
+        return nullptr;
+    }
+
+    if (!value.isMember("Cmd") || !value["Cmd"].isString())
+    {
+        ChipLogError(NotSpecified, "Lock App: Invalid JSON command received: command name is missing");
+        return nullptr;
+    }
+
+    Json::Value params = Json::objectValue;
+    if (value.isMember("Params"))
+    {
+        if (!value["Params"].isObject())
+        {
+            ChipLogError(NotSpecified, "Lock App: Invalid JSON command received: specified parameters are incorrect");
+            return nullptr;
+        }
+        params = value["Params"];
+    }
+    auto commandName = value["Cmd"].asString();
+    return new LockAppCommandHandler(commandName, params);
```

Comment:
we should not use new directly in Matter, use Platfrom::New

Reply:
Fixed, thanks.